docs/architecture/adr-010-remove-wire-msg-pay-for-blob.md

```diff
@@ -0,0 +1,99 @@
+# ADR 010: Remove `WireMsgPayForBlob`
+
+## Changelog
+
+- 2022-11-14: Initial draft
+
+## Context
+
+### Cosmos SDK transactions vs messages
+
+For the remainder of this document, sdk.Tx refers to a Cosmos SDK transaction and sdk.Msg refers to a Cosmos SDK message.
+
+- A single sdk.Tx may contain one or many sdk.Msg
+- A sdk.Tx's `ValidateBasic()` is distinct from a sdk.Msg's `ValidateBasic()`
+
+### `WireMsgPayForBlob` vs `MsgPayForBlob`
+
+Historically, `WireMsgPayForBlob` was needed so that a user could create, sign, and send multiple signatures per message share commitment (one per square size). The user would submit their `WireMsgPayForBlob` as the sole message in a transaction to the mempool where a block proposer could pick it, malleate it into a `MsgPayForBlob` that included the appropriate signature for the block being constructed and include it in a block.
+
+With the introduction of [ADR 008: square size independent message commitments](./adr-008-square-size-independent-message-commitments.md), a user no longer needs to create, sign, and send multiple signatures. This enables us to reduce the complexity of the malleation process by removing `WireMsgPayForBlob` entirely. Instead, users will create and publish a `BlobTx` to the mempool. The `BlobTx` will include a `sdk.Tx` which will remain unmodified and end up on-chain.
+
+// TODO: diagram the transaction flow for a BlobTx from user -> block proposer -> tx in a block
+
+### `MalleatedTx`'s OriginalTxHash
+
+celestia-core contains a patch to replace the tx hash of a tx containing a `MsgPayForBlob` with the tx hash from the original `WireMsgPayForBlob` it was derived from. This change was needed because the transaction the user creates (one containing a single `WireMsgPayForBlob`) will always be different from the transaction that is included in a block (one containing a single `MsgPayForBlob`). Since the transactions are different, the hashes will also be different. This means Tendermint's default transaction indexing can't confirm to a user that their transaction was included in a block.
+
+See [celestia-core#607](https://github.com/celestiaorg/celestia-core/pull/607) and Tendermint docs on [indexing transactions](https://docs.tendermint.com/v0.34/app-dev/indexing-transactions.html).
+
+## Alternative Approaches
+
+Preserve existing `WireMsgPayForBlob` and `MsgPayForBlob`.
+
+## Decision
+
+Proposed
+
+## Detailed Design
+
+1. In celestia-core, introduce a new Protobuf definition for `BlobTx`
+
+    ```proto
+    // BlobTx wraps an encoded sdk.Tx with a second field to contain blobs of
+    // data to be published to the Celestia blockchain. The raw bytes of the
+    // blobs are not signed over but they are verified using the tx
+    // MsgPayForBlobs's share commitments which are signed over.
+    message BlobTx {
+        bytes tx = 1; // marshalled sdk.Tx of type MsgPayForBlobs
```

Comment:
I think we need to define `MsgPayForBlobs` in this ADR because the current types defined in [celestia-app/tx.proto](https://github.com/celestiaorg/celestia-app/blob/24626ff8d00854d7e0027abcfdf50de1a2aee6a3/proto/blob/tx.proto) aren't sufficient if we intend on supporting multiple blobs in a message.

```proto
message MsgPayForBlob {
  string signer = 1;
  repeated bytes namespace_ids = 2;
  repeated uint64 blob_sizes = 3;
  repeated bytes share_commitments = 4;
  repeated uint32 share_versions = 8;
}
```

but I think we should adopt an incremental approach to this and support a singular message with a singular blob first. In that scenario, we can use 

```proto
message MsgPayForBlob {
  string signer = 1;
  bytes namespace_id = 2;
  uint64 blob_size = 3;
  bytes share_commitment = 4;
  uint32 share_version = 8;
}
```